os/drivers/lcd/mipi_lcd.c

```diff
@@ -349,5 +424,17 @@ FAR struct lcd_dev_s *mipi_lcdinitialize(FAR struct mipi_dsi_device *dsi, struct
 	}
 	priv->config->backlight(CONFIG_LCD_MAXPOWER);
 
+#if defined(CONFIG_LCD_ST7701SN)
```

Comment:
It should be same condition as in putarea function, no need for malloc in portrait 
 #if defined(CONFIG_LCD_ST7701SN) && (defined(CONFIG_LCD_LANDSCAPE) || defined(CONFIG_LCD_RLANDSCAPE))

Reply:
done